### ISSUE TITLE
[BUG] Local hosted LM Studio AI 'model_dump_json' error

### ISSUE THREAD

**Issue by Renno231** [browser-use/browser-use#75]
using try.py from examples with base_url set to LM Studio
AttributeError: 'NoneType' object has no attribute 'current_state'

**Comment by Hydrapralix** [comment]
Same here with ollama using 
`from asyncio import run
from browser_use import Agent
from langchain_ollama import ChatOllama

llm = ChatOllama(
    model="llama3-groq-tool-use:8b",
    base_url="http://localhost:11434",
    
)

agent = Agent(
    task="Go to google.com and search for 'Albert Einstein'.",
    llm=llm,
)

run(agent.run())`

**Comment by descention** [comment]
The output of the LLM isn't including the full json response as defined in [the prompt](https://github.com/gregpr07/browser-use/blob/7d7659cfa6d4a2d415c80eab20d4f302eddb1ca5/browser_use/agent/prompts.py#L23C1-L38C5). If you want to debug it, check what your LLM response looks like and try adjusting the prompt to get it to respond with the correct json properties.

I've tried, without success:
* llama3.2:3b-instruct-q8_0
* qwen2.5-coder:7b-instruct-q6_K
* phi3:3.8b-mini-128k-instruct-q8_0

**Comment by jfitz001** [comment]
Yeah i have tried 6 different vision LLMs with LM Studio and none of them seem to work. Most of the fail like so
```
INFO     [agent] 
📍 Step 1
ERROR    [agent] ❌ Result failed 1/5 times:
 'NoneType' object has no attribute 'action'
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 2/5 times:
 'NoneType' object has no attribute 'action'
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 3/5 times:
 'NoneType' object has no attribute 'action'
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 4/5 times:
 'NoneType' object has no attribute 'action'
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 5/5 times:
 'NoneType' object has no attribute 'action'
ERROR    [agent] ❌ Stopping due to 5 consecutive failures
```

**Comment by Mohit-majumdar** [comment]
I have tried using locally hosted models to get it to work, but I keep getting the error *'NoneType' object has no attribute 'action'*.
Is there a way to resolve this? 

i am using llama3.1

**Comment by descention** [comment]
> I have tried using locally hosted models to get it to work, but I keep getting the error *'NoneType' object has no attribute 'action'*.
> 
> Is there a way to resolve this? 
> 
> 
> 
> i am using llama3.1

Convince the model to use the correct response schema.

**Comment by csharpmastr** [comment]
I think the models to be used should have vision capabilities. So a text-image-text models should work just fine.

**Comment by anhtester** [comment]
Did you solve it? Please show me which llm package to use.
Thank you!!

**Comment by Cfomodz** [comment]
This particular error is new, but as pointed out by @descention this is an issue with [instruction following](https://paperswithcode.com/task/instruction-following)  and is tied directly to the model being used, not Browser-use, at least technically speaking.

While I could offer some troubleshooting steps for LM Studio,  getting it to run without an error once is one thing, but having it actually be able to drive BU consistently on non-arbitrary tasks with actual context being needed (making it respond with JSON that meets spec less and less) and that is why I have not pushed for us to have such troubleshooting tips in the docs.

For general model setup and supported configurations, please refer to our documentation here:
- [Supported Models](https://github.com/browser-use/browser-use/blob/main/docs/customize/supported-models.mdx)
- [Local Development & Setup](https://github.com/browser-use/browser-use/blob/main/docs/development/local-setup.mdx)

And with that said, I always recommend having a working cloud model to compare results against. If 4o is throwing an error, then it's not just your model. If it just errors on your system but works fine on cloud or with a cloud based LLM via API (self hosted), then it's most likely just due to the prompt and required response formatting not being followed, or, just straight up not being possible at the model size you are trying it with.

We’re actively working on a more robust benchmarking framework to provide more granular insights into how both cloud and local models (including various size options) perform with browser-use.

If this issue was misunderstood, closed in error, or if you have more details to add, we apologize—please feel free to follow up or create a new issue with additional context. Your feedback is valued and helps us improve both the project and the docs.
